sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/filter/Predicate.java

```diff
@@ -145,5 +151,8 @@ public class Predicate extends GeneralScalarExpression {
 
   public Predicate(String name, Expression[] children) {
     super(name, children);
+    if ("BOOLEAN_EXPRESSION".equals(name)) {
```

Comment:
I think we should use `final` to modify "BOOLEAN_EXPRESSION".

Reply:
all other functions names are hardcoded and we just match the string literal in the code.